SimTKcommon/tests/TestPlugin.cpp

```diff
@@ -190,6 +190,488 @@ void testPathname() {
     SimTK_TEST(!isAbsPath
         && directory=="topdir"+sep+"seconddir"+sep
         && fileName=="myFileName" && extension==".ext");
+
+#ifdef _WIN32
+    std::string swd, path, dir, pathname;
+    bool dontApplySearchPath;
+    const std::string cwd = Pathname::getCurrentWorkingDirectory();
+    std::string cwd_nodrive = cwd; cwd_nodrive.erase(0,3);
+    const std::string cwdX = Pathname::getCurrentWorkingDirectory("x");
+    const std::string cwdY = Pathname::getCurrentWorkingDirectory("y");
+
+    swd = ""; path = "X:/topdir/seconddir/myFileName.ext";
+    dir = "x:" + sep + "topdir" + sep + "seconddir" + sep;
+    directory = fileName = extension = "junk"; dontApplySearchPath = false;
+    Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(swd, path, directory, fileName, extension, dontApplySearchPath);
+    SimTK_TEST(directory == dir && fileName == "myFileName" && extension == ".ext" && dontApplySearchPath);
+    pathname = "junk";
```

Comment:
You don't need these `pathname="junk"` lines because pathname is overridden by the next line. That could only fail if the compiler is broken -- it doesn't matter what the routine does.